pdr-api/handlers/fees/PUT/index.js

```diff
@@ -0,0 +1,147 @@
+    }
+
+    // If no currentRecord exists, we shouldn't perform any actions
+    const currentRecord = await getOne(pk, sk);
+    if (!currentRecord?.pk) {
+      logger.error(`Protected area with identifier '${queryParams.ORCS}' not found.`);
+      return sendResponse(400, {}, 'Bad Request', `Protected area with identifier '${queryParams.ORCS}' not found.`, context);
+    }
+
+    // Call the 'updateRecord' function to update the fee
+    let attributes = await updateRecord(pk, sk, body);
+    return sendResponse(200, attributes, 'Record updated');
+  } catch (error) {
+    // Log any caught errors
+    logger.error('Error:', error);
+    return sendResponse(400, [], error?.msg || 'Error', error?.error || error, context);
+  }
+};
+
+/**
+ * Updates a record in DynamoDB.
+ *
+ * @param {String} pk
+ * @param {String} sk
+ * @param {Object} body - The request body.
+ * @returns {Promise<Object>} - A Promise that resolves to the updated attributes.
+ */
+async function updateRecord(pk, sk, body) {
+  let attributeName;
+  let expressionAttributeNames = {};
+  let updatedAttributeValues = {};
+  let updateExpression = [];
+
+  for (const field of OPTIONAL_PUT_FIELDS) {
+    if (body.hasOwnProperty(field)) {
+      attributeName = `#${field}`;
+      expressionAttributeNames[attributeName] = field;
+      updatedAttributeValues[`:${field}`] = { S: body[field] || '' };
+      updateExpression.push(`${attributeName} = :${field}`);
+    }
+  }
+
+  const updateParams = {
+    TableName: TABLE_NAME,
+    Key: {
+      pk: { S: pk },
+      sk: { S: sk }
+    },
+    ExpressionAttributeNames: expressionAttributeNames,
+    ExpressionAttributeValues: updatedAttributeValues,
+    UpdateExpression: `SET ${updateExpression.join(', ')}`,
+    ReturnValues: 'ALL_NEW'
+  };
+
+  try {
+    
+    // Update the item
+    logger.debug('updateParams', updateParams);
+    const { Attributes } = await dynamodb.updateItem(updateParams);
+
+    // success unmarshall
+    logger.debug('Update success:', unmarshall(Attributes));
+    return unmarshall(Attributes);
+
+  } catch (error) {
+    // Log any errors that occur during the update operation
+    logger.error(error);
+    let conditionalErrorFlag = false;
+    if (error?.CancellationReasons) {
+      // Check for ConditionalCheckFailed with transactional update
+      conditionalErrorFlag = error.CancellationReasons.find(item => {
+        if (item?.Code === 'ConditionalCheckFailed') {
+          return true;
+        }
+        return false;
+      });
+    }
+    if (error?.code === 'ConditionalCheckFailedException') {
+      // Check for ConditionalCheckFailedException with single item update
+      conditionalErrorFlag = true;
+    }
+    if (conditionalErrorFlag) {
+      throw `Field mismatch: Error during update.`;
+    }
+    throw error;
+  }
+}
+
+// Acceptible fields
+const OPTIONAL_PUT_FIELDS = [
+  'Night',
+  'Day',
+  'Use',
+  'Week',
+  'Year',
+  'Trip',
+  'DirectionOfTrip',
+  'Days28'
+];
```

Comment:
unless exported, should be declared at the very top of the file, or even better in a re-usable layer const that is exported.

Reply:
Addressed in : [Commit](79210c7e569ed7e600615bc738cc5889c18bc619)